lib/cocoapods-spm/hooks/post_integrate/1.add_spm_pkgs.rb

```diff
@@ -8,13 +8,32 @@ class AddSpmPkgs < Hook
         def run
           return if @spm_resolver.result.spm_pkgs.empty?
 
-          add_spm_pkg_refs_to_project
-          add_spm_products_to_targets
-          pods_project.save
+          if podfile.installation_options.generate_multiple_pod_projects
```

Comment:
to simplify the logic, how about this way (not having if/else check):
```ruby
projects_to_integrate.each do |project|
  add_spm_pkg_refs_to_project(project)
  add_spm_products_to_targets(project)
  project.save
end
```
where `projects_to_integrate` is:
```ruby
# in base.rb
def projects_to_integrate
  [pods_project] + @context.pod_target_subprojects
end
```

Reply:
it's great, i will have a try.

Reply:
pods_project should be saved last, or project info could be write into disk

Reply:
after execute as suggestion above, every pod_target_subproject will dep all spms in there root_object and targets, which may not be correct

Reply:
today was will try to refine the code as you suggest, maybe do some Chang

Reply:
I refined the spms add logic, add spm_ref to targets of project first, and then add the  pgk_refs to the project.